repose-aggregator/components/filters/uri-stripper-filter/src/test/scala/org/openrepose/filters/uristripper/UriStripperFilterRequestTest.scala

```diff
@@ -0,0 +1,1028 @@
+            |    </link-resource>
+            |</uri-stripper>
+          """.stripMargin
+
+        val requestBody =
+          """{
+            |  "link": "http://example.com/v1/bar/12345"
+            |}
+          """.stripMargin
+
+        filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+        request.setRequestURI("/v1/12345/foo")
+        request.setContent(requestBody.getBytes(CHARSET_UTF8))
+        request.setContentType(MimeType.APPLICATION_JSON.toString)
+
+        filter.doFilter(request, response, filterChain)
+
+        (Json.parse(getPostFilterRequestBody) \ "link").as[String] shouldEqual "http://example.com/v1/bar"
+      }
+
+      it("should not alter the body if the link cannot be located (continue)") {
+        val config =
+          """<?xml version="1.0" encoding="UTF-8"?>
+            |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+            |    <link-resource uri-path-regex=".*">
+            |        <request>
+            |            <json link-mismatch-action="continue">$.dne</json>
+            |        </request>
+            |    </link-resource>
+            |</uri-stripper>
+          """.stripMargin
+
+        val requestBody =
+          """{
+            |  "link": "http://example.com/v1/12345/foo"
+            |}
+          """.stripMargin
+
+        filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+        request.setRequestURI("/v1/12345/foo")
+        request.setContent(requestBody.getBytes(CHARSET_UTF8))
+        request.setContentType(MimeType.APPLICATION_JSON.toString)
+
+        filter.doFilter(request, response, filterChain)
+
+        getPostFilterRequestBody shouldEqual Json.parse(requestBody).toString()
+      }
+
+      it("should not alter the body if the link's token index is out of bounds (continue)") {
+        val config =
+          """<?xml version="1.0" encoding="UTF-8"?>
+            |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+            |    <link-resource uri-path-regex=".*">
+            |        <request>
+            |            <json link-mismatch-action="continue" token-index="5">$.link</json>
+            |        </request>
+            |    </link-resource>
+            |</uri-stripper>
+          """.stripMargin
+
+        val requestBody =
+          """{
+            |  "link": "http://example.com/v1/12345/foo"
+            |}
+          """.stripMargin
+
+        filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+        request.setRequestURI("/v1/12345/foo")
+        request.setContent(requestBody.getBytes(CHARSET_UTF8))
+        request.setContentType(MimeType.APPLICATION_JSON.toString)
+
+        filter.doFilter(request, response, filterChain)
+
+        getPostFilterRequestBody shouldEqual Json.parse(requestBody).toString()
```

Comment:
Maybe the tests shouldn't say `not alter`, but rather `not alter meaning`. During the processing, all unnecessary white space is removed including the `\n`'s in the `requestBody` value. So the `Json.parse().toString` accomplishes the same thing in a generic way for the test comparisons.